### ISSUE TITLE
Discrepancy in Llava-cot Results Between Paper and VLMEvalKit Leaderboard

### ISSUE THREAD

**Issue by JCruan519** [PKU-YuanGroup/LLaVA-CoT#18]
Hello,

Thank you for your valuable work!

I noticed in your paper that the evaluation tool used is VLMEvalKit. However, on the VLMEvalKit leaderboard (https://rank.opencompass.org.cn/leaderboard-multimodal/?m=REALTIME), the reported test results for Llava-cot differ significantly from those presented in your paper.

I hope you can help address my concerns. Thank you!

![](https://github.com/user-attachments/assets/1b9f1185-71c6-4016-86d4-1342f626d379)
![](https://github.com/user-attachments/assets/c65062dc-58b2-4a1d-9380-7eaae174ec31)

**Comment by XuGW-Kevin** [comment]
Hi @JCruan519 ,
Thank you so much for raising this issue! If the difference is within 1-2%, it might be because they did not filter out the content between <CONCLUSION></CONCLUSION>. However, regardless of the testing method, such a significant discrepancy is impossible. The result reported by OpenCompass is clearly unreasonable. Thank you again for bringing this to our attention, and we will follow up on this matter.

**Comment by XuGW-Kevin** [comment]
We provide an evidence: https://huggingface.co/spaces/opencompass/Open_LMM_Reasoning_Leaderboard.
This is also an OpenCompass leaderboard. On this leaderboard, as of the release date, our model ranked second among 10B-scale models, only behind Qwen2-VL-7B, and performed significantly better than Llama-3.2V-11B. However, on the leaderboard you provided, our model performs even significantly worse than Llama-3.2V-11B, which we believe is impossible.

**Comment by XuGW-Kevin** [comment]
@JCruan519 We have basically identified the issue: OpenCompass used the Llama-3.2V configuration when running the tests, which sets max_token to 128/512. This is completely insufficient for our model (2048), especially for challenging tasks. This explains why our model performed significantly worse, particularly on tasks like MMMU.

**Comment by JCruan519** [comment]
@XuGW-Kevin 
Hello, thank you very much for your response!

In the link you provided, it seems that only math-related benchmarks are included, with MathVista being the repeated benchmark in your paper. The result in the provided link is 54.2, in your paper it is 54.8, and in the link I provided it is 53.5. These are quite similar, and I don't have any concerns about them.

However, for MMStar, MMBench, HallusionBench, and AI2D, the results show significant differences.

Additionally, during my local reproduction using VLMEvalKit, I found that, for example, in MMStar, due to the influence of the max_new_tokens parameter, and because Llava-COT outputs relatively long reasoning processes, the results are truncated before they are fully generated. This may be one of the reasons for the large discrepancies in performance.

Therefore, I would greatly appreciate it if you could provide the evaluation setup details for VLMEvalKit, such as the max_new_tokens parameter!

**Comment by XuGW-Kevin** [comment]
Thank you very much for your willingness to test locally! In fact, we have provided the evaluation parameters and instructions for modifying VLMEvalKit at https://huggingface.co/Xkev/Llama-3.2V-11B-cot. However, it seems that many users still tend to use the original parameters.

**The modification involves replacing self.kwargs with:
dict(do_sample=True, temperature=0.6, top_p=0.9, max_new_tokens=2048).**

I have already reported this to OpenCompass, and they mentioned they would re-run the tests. Just now, I immediately ran a test on MMStar (this is why our response was slightly delayed), and the result was 56.7% (complete answers) and 57.6% (answers only within <CONCLUSION></CONCLUSION>),  which is exactly consistent with the reported 57.6% (answers only within <CONCLUSION></CONCLUSION>).

We provide the generated results here for your reference:
complete answers: [base_model_MMStar.xlsx](https://github.com/user-attachments/files/18222376/base_model_MMStar.xlsx)
answers only within <CONCLUSION></CONCLUSION>: [base_model_MMStar.xlsx](https://github.com/user-attachments/files/18222385/base_model_MMStar.xlsx)

To facilitate verifying our analysis locally, I will soon upload an evaluation script that you can download and run on your local setup. Thank you very much for your support!

**Comment by JCruan519** [comment]
@XuGW-Kevin 
Thank you very much for your timely reply! 
I indeed only focused on the GitHub homepage and the paper, neglecting Huggingface homepage. I will adjust the parameter settings and conduct local testing later. 
Once again, thank you for your timely response!

**Comment by JCruan519** [comment]
Hello! I used the evaluation.txt you provided above to retest Llava-cot locally. The test results on the other four benchmarks show minimal deviation from the original paper. However, on AI2D, my local test results show an overall score of only 77.49. On MMVet, the result is only 27.94, both of which differ significantly from the original paper. I have attached my XLSX file for your reference:

[LLaVA-CoT_AI2D_TEST_openai_result.xlsx](https://github.com/user-attachments/files/18243157/LLaVA-CoT_AI2D_TEST_openai_result.xlsx)
[LLaVA-CoT_MMVet_gpt-4-turbo.xlsx](https://github.com/user-attachments/files/18243160/LLaVA-CoT_MMVet_gpt-4-turbo.xlsx)

I would greatly appreciate it if you could help resolve my doubts!

**Comment by XuGW-Kevin** [comment]
Hi, we’re glad you were able to take the time to reproduce our work so quickly!

1. MMVet Benchmark
![image](https://github.com/user-attachments/assets/6946d8d5-c7a8-452c-8033-30016fd8811e)
This is likely a network issue. (In many examples, we answer correctly, but GPT scores 0.)
**Note that Opencompass also reports 63.6% on this benchmark.**

2. AI2D Benchmark
In fact, since you raised this issue three days ago, we have conducted additional tests. Previously, we had been using the AI2D_TEST_NO_MASK version of VLMEvalKit. **As a result, the AI2D portion of our training dataset was also trained using the NO_MASK images.**
After you brought up this issue, we carefully tried to reproduce why our results differed from those of OpenCompass. We discovered that they were using the AI2D_TEST dataset in their benchmark, not AI2D_TEST_NO_MASK.  (To the best of my knowledge, they did not mention this anywhere. I thought these two versions have no significant difference.) We realized this was an error we had overlooked previously. **As a result, we are now replacing the AI2D images in our training set with the MASK versions, retraining the model, and updating the data in our paper.** We will report four numbers: train on NO_MASK, test on MASK/NO_MASK; train on MASK, test on MASK/NO_MASK.

**Comment by JCruan519** [comment]
Got it, thank you for your quick response! I will check my judge API again.

**Comment by yxchng** [comment]
just curious, why use test data from ai2d (whether no_mask or mask) in the training data? do other models also do the same? isn't this unfair?

**Comment by XuGW-Kevin** [comment]
> just curious, why use test data from ai2d (whether no_mask or mask) in the training data? do other models also do the same? isn't this unfair?

Hi, different from other benchmarks, AI2D has its training dataset, I know what we used is in fact the training dataset. Other leading models (InternVL etc.) also used the training dataset. Using the test dataset is of course unfair.

**Comment by yxchng** [comment]
where can i find the original training dataset? i can't seem to find it on the official site

**Comment by XuGW-Kevin** [comment]
The training dataset is here: https://huggingface.co/datasets/Xkev/LLaVA-CoT-100k
(We included this in the "News" part.)
